components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/messaging_reference.adoc

```diff
@@ -0,0 +1,589 @@
+* xref:#MessageToL2CancellationStarted[`MessageToL2CancellationStarted`]
+
+|===
+
+== L1 function reference
+
+Functions are listed in alphabetical order.
+
+'''
+
+[#cancelL1ToL2Message]
+=== `cancelL1ToL2Message`
+
+[discrete]
+==== Description
+
+Cancels an L1 to L2 message. Call this function after calling the xref:#startL1ToL2MessageCancellation[`startL1ToL2MessageCancellation`] function. The time between the calls to these two functions must be at least the number of seconds defined by the xref:#messageCancellationDelay[`messageCancellationDelay`] function.
+
+Only a sender can cancel a message.
+
+If the message is missing, the call reverts.
+
+Be aware that the message fee is not refunded.
+
+[discrete]
+==== State Mutability
+
+None.
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _toAddress_`:: The address of the L2 contract.
+`uint256 _selector_`::
+// tag::uint256_selector[]
+The function, in the recipient L2 contract, that the message called.
+// end::uint256_selector[]
+`uint256[] calldata _payload_`:: The payload of the message.
+`uint256 _nonce_`:: The nonce of the message.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The hash of the canceled message.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L161[`cancelL1ToL2Message`]
+
+'''
+
+[#consumeMessageFromL2]
+=== `consumeMessageFromL2`
+
+[discrete]
+==== Description
+
+Consumes a message that was sent from an L2 contract.
+
+Returns the hash of the message.
+
+[discrete]
+==== State Mutability
+
+None.
+
+[discrete]
+==== Parameters
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`uint256 _fromAddress_`:: The address of the L2 contract sending the message.
+`uint256[] calldata _payload_`:: The payload of the message.
+
+[discrete]
+==== Returns
+
+[horizontal,labelwidth="30",role=stripes-odd]
+`bytes32 _msgHash_`:: The hash of the consumed message.
+
+[discrete]
+==== Function definition
+
+Contract: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol[`StarknetMessaging.sol`]
+
+* Function: link:https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/solidity/StarknetMessaging.sol#L132[`consumeMessageFromL2`]
+
+'''
+
+[#getMaxL1MsgFee]
+=== `getMaxL1MsgFee`
+
+[discrete]
+==== Description
+
+Returns the maximum fee, in Wei, that Starknet #charges# #(Is "charges" correct? It says "accepts" link:https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/starknet/solidity/IStarknetMessaging.sol#L8[in the source file])# for a single message.
```

Comment:
@LucasLvy Is "charges" correct? It says "accepts" [in the source file](https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/starknet/solidity/IStarknetMessaging.sol#L8).

Reply:
I spoke with Remo and changed accordingly.